sycl/include/CL/cl_usm_ext.h

```diff
@@ -52,7 +52,7 @@ typedef cl_uint cl_mem_info_intel;
 #define CL_MEM_ALLOC_TYPE_INTEL 0x419A
 #define CL_MEM_ALLOC_BASE_PTR_INTEL 0x419B
 #define CL_MEM_ALLOC_SIZE_INTEL 0x419C
-#define CL_MEM_ALLOC_DEVICE_INTEL 0x419D
+#define CL_MEM_ALLOC_INFO_DEVICE_INTEL 0x419D
```

Comment:
Intel already uses the ID https://github.com/intel/compute-runtime/blob/master/public/cl_ext_private.h#L103
Shouldn't we change the ID or align on name?

Reply:
Will revert and get @bashbaug and @MichalMrozek to fight it out

Reply:
`CL_MEM_ALLOC_DEVICE_INTEL` is the name I have been using.

Did you see `CL_MEM_ALLOC_INFO_DEVICE_INTEL` somewhere?  If so, we should get that fixed to avoid possible confusion in the future.  FWIW, I've checked the specs and the Intercept Layer and I think they're both using `CL_MEM_ALLOC_DEVICE_INTEL` exclusively.

Reply:
It is introduced in https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/USM/cl_intel_unified_shared_memory.asciidoc

It is resaonable to change it there as well as it is in the same repo.